src/d3a/models/market/market_structures.py

```diff
@@ -177,17 +177,40 @@ def offer_or_bid_from_JSON_string(offer_or_bid, current_time):
         return Bid(**offer_bid_dict)
 
 
-class TradeBidInfo(namedtuple('TradeBidInfo',
-                              ('original_bid_rate', 'propagated_bid_rate',
-                               'original_offer_rate', 'propagated_offer_rate',
-                               'trade_rate'))):
+def _is_offer(offer_or_bid):
+    offer_bid_dict = json.loads(offer_or_bid)
+    object_type = offer_bid_dict.pop("type")
+    if object_type == "Offer":
+        return True
+    else:
+        return False
```

Comment:
Minor, but this could be simplified as:
```suggestion
    return object_type == "Offer":
```
(also in _is_bid)

Reply:
Right, that's better